cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -643,41 +641,25 @@ func (c *CStorVolumeReplicaController) populateVersion(cvr *apis.CStorVolumeRepl
 			Update(cvr)
 
 		if err != nil {
-			return nil, err
+			return cvr, err
 		}
 		klog.Infof("Version %s added on cvr %s", v, cvr.Name)
 		return obj, nil
 	}
 	return cvr, nil
 }
 
-func isCurrentVersionValid(cvr *apis.CStorVolumeReplica) bool {
-	validVersions := []string{"1.0.0", "1.1.0", "1.2.0"}
-	version := strings.Split(cvr.VersionDetails.Status.Current, "-")[0]
-	return util.ContainsString(validVersions, version)
-}
-
-func isDesiredVersionValid(cvr *apis.CStorVolumeReplica) bool {
-	validVersions := []string{"1.3.0"}
-	version := strings.Split(cvr.VersionDetails.Desired, "-")[0]
-	return util.ContainsString(validVersions, version)
-}
-
-func upgradePath(cvr *apis.CStorVolumeReplica) string {
-	return strings.Split(cvr.VersionDetails.Status.Current, "-")[0] + "-" +
-		strings.Split(cvr.VersionDetails.Desired, "-")[0]
-}
-
 func setReplicaID(u *upgradeParams) (*apis.CStorVolumeReplica, error) {
 	cvr := u.cvr
 	err := volumereplica.GetAndUpdateReplicaID(cvr)
 	if err != nil {
-		return nil, err
+		return cvr, err
 	}
-	cvr, err = u.client.OpenebsV1alpha1().
-		CStorVolumeReplicas(cvr.Namespace).Update(cvr)
+	cvrObj := cvr.DeepCopy()
```

Comment:
there is no need here.. if we need, it should be before calling GetAndUpdateReplicaID..
without having is also fine as there are no checks for reconcilation over spec.ReplicaID. If there are any checks, then, we need DeepCopy before calling GetAndUpdateReplicaID

Reply:
Moved it up before GetAndUpdateReplicaID